Задача: Кэшируем, кэшируем.

piligrim-curious opened the pull request:
**WeatherCacheTemplate:** Абстрактный класс;
**FrequentUpdateWeatherCache**: Класс с частым обновлением;
**StandardWeatherCache**: Класс с обычным режимом кэширования;
**WeatherData**: Класс данных о погоде;
**WeatherService**: Класс получения данных о погоде;
**WeatherProvider**: Интерфейс;
**WeatherCacheTest**: Юнит-тесты.

piligrim-curious committed:
Задача: Кэшируем, кэшируем.
WeatherCacheTemplate: Абстрактный класс;
FrequentUpdateWeatherCache: Класс с частым обновлением;
StandardWeatherCache: Класс с обычным режимом кэширования;
WeatherData: Класс данных о погоде;
WeatherService: Класс получения данных о погоде;
WeatherProvider: Интерфейс;
WeatherCacheTest: Юнит-тесты.

SnurnicynKonstantin commented on src/main/java/school/faang/cash/cash/WeatherData.java:
```diff
@@ -0,0 +1,15 @@
+@Getter
+@ToString
+@AllArgsConstructor
+public class WeatherData {
```
Попробуй на record переделать вместо class

SnurnicynKonstantin commented on src/main/java/school/faang/cash/cash/WeatherService.java:
```diff
@@ -0,0 +1,21 @@
+    private final Random random = new Random();
+
+    public WeatherData fetchWeatherData(String city) {
+        double minTemp = -10.0;
```
Эти две переменные надо в константы вынести

SnurnicynKonstantin commented on src/main/java/school/faang/cash/cash/WeatherService.java:
```diff
@@ -0,0 +1,21 @@
+        double temperature = Math
+                .round((minTemp + (maxTemp - minTemp) * random.nextDouble()) * 10.0) / 10.0;
+
+        double minHumidity = 10;
```
И эти в константы

SnurnicynKonstantin commented on src/test/java/school/faang/cash/cash/WeatherCacheTest.java:
```diff
@@ -0,0 +1,115 @@
+    private StandardWeatherCache standardCache;
+    private FrequentUpdateWeatherCache frequentCache;
+
+    @BeforeEach
```
Вау, и тут тесты пишешь. Не перестаешь удивлять)

piligrim-curious committed:
Задача: Кэшируем, кэшируем.
WeatherCacheTemplate: Абстрактный класс;
FrequentUpdateWeatherCache: Класс с частым обновлением;
StandardWeatherCache: Класс с обычным режимом кэширования;
WeatherData: Класс данных о погоде;
WeatherService: Класс получения данных о погоде;
WeatherProvider: Интерфейс;
WeatherCacheTest: Юнит-тесты.

Изменения:
(WeatherData.java)
* class -> record;
(WeatherService.java)
* Все переменные с фиксированными значениями были вынесены в константы;
(Общие)
* Все геттеры были упрощены до названия полей класса.

SnurnicynKonstantin reviewed (approved):
Теперь на константах и рекордах еще аккуратней получилось.
